libyul/AsmParser.cpp

```diff
@@ -78,16 +64,77 @@ unique_ptr<Block> Parser::parse(std::shared_ptr<Scanner> const& _scanner, bool _
 	return nullptr;
 }
 
+bool Parser::updateLocation()
+{
+	m_documentedLocation.reset();
+	if (m_scanner->currentCommentLiteral().empty())
+		return false;
+
+	for (auto const [name, value, ok]: splitDoxyStyleComments(m_scanner->currentCommentLiteral()))
+	{
+		if (!ok)
+			break;
+
+		if (name == "use-src"sv)
+		{
+			// Don't handle that for now.
+		}
+		else if (name == "src"sv)
+		{
+			// Value is of form: <sourceIndex>:<start>,<end>
+			regex const sourceMatchSyntax(
+				R"~~~(^(\d+):(-1|\d+):(-1|\d+)$)~~~",
+				std::regex_constants::ECMAScript | std::regex_constants::optimize
+			);
+
+			cmatch sm;
+			regex_search(value.data(), value.data() + value.size(), sm, sourceMatchSyntax);
+
+			solAssert(sm.size() == 4, "");
+			auto const sourceIndex = static_cast<unsigned>(stoul(sm[1].str()));
+			auto const start = stoi(sm[2].str());
+			auto const end = stoi(sm[3].str());
+
+			solAssert(!!m_charStreamForSourceIndex, "");
+			auto charStream = m_charStreamForSourceIndex(sourceIndex);
+			solAssert(charStream, "");
+
+			m_documentedLocation = SourceLocation{start, end, charStream};
+			return true;
+		}
+		// otherwise, ignore unknown keys for now
+	}
+	return false;
+}
+
+[[nodiscard]]
+shared_ptr<DebugData const> Parser::updateLocationEndFrom(
+	shared_ptr<DebugData const> const& _debugData,
+	langutil::SourceLocation const& _location,
+	bool _fromDoc
```

Comment:
Shouldn't `_fromDoc` be a switch that is activated once in the parser when the first tag it sees is `@use-src` and disabled otherwise?

Reply:
I think it is possible to have source locations mixed. I think when you parse yul code some parts can be solidity transformed yul code, and some auto-generated or injected, isn't it?

Reply:
IIRC we agreed to not allow mixed modes for now. the other is implemented. hiding & marking as resolved.